Source/Core/Core/HW/SI.cpp

```diff
@@ -234,9 +234,9 @@ void DoState(PointerWrap &p)
 		{
 			// If no movie is active, we'll assume the user wants to keep their current devices
 			// instead of the ones they had when the savestate was created.
-			if (!Movie::IsMovieActive())
-				return;
-
+			// However, change the devices anyways in case they add or remove
+			// a device after the savestate was created.
+			ChangeDeviceDeterministic(type, i);
```

Comment:
And this call needs to be done after you have added the saved stated type (save the result of device->GetDeviceType()), otherwise ChangeDeviceDetermintice will be a no-op.